lib/roby/custom_require.rb

```diff
@@ -1,27 +1,52 @@
 # frozen_string_literal: true
 
+require "pathname"
+
 # This is needed because of Ruby bug #9244
-module Kernel
-    alias syskit_original_require require
-    def require(path)
-        # Only deal with paths starting with models/, the other ones are
-        # unambiguous
-        if path =~ /^models\//
-            Roby.app.search_path.each do |app_dir|
-                full_path = File.join(app_dir, path)
-                if File.file?(full_path) || File.file?("#{full_path}.rb")
-                    return syskit_original_require(full_path)
-                end
+module Syskit
+    module CustomRequire #:nodoc:
+        def self.resolve(path)
+            match = /^([\w|-]+)\//.match(path)
+            return path unless match
+
+            prefix = match[1]
+            if (full_path = resolve_from_apps(prefix, path))
+                return full_path
             end
-        elsif (match = /^(\w+)\/models\//.match(path))
-            app_name = match[1]
-            if (base_dir = Roby.app.find_registered_app_path(app_name))
-                full_path = File.join(File.dirname(base_dir), path)
-                if File.file?(full_path) || File.file?("#{full_path}.rb")
-                    return syskit_original_require(full_path)
-                end
+
+            if (full_path = resolve_from_search_path(path))
+                return full_path
             end
+
+            path
```

Comment:
A FYI, this is a matter of taste, I would have done

~~~
resolve_from_apps(prefix, path) ||
  resolve_from_search_path(path) ||
  path
~~~

Reply:
I liked it, I'll follow that on a next time